lib/src/convert_es6_proxy.html

```diff
@@ -0,0 +1,198 @@
+
+          if (arrayCommonBehaviors.hasOwnProperty(propertyName)) {
+            return arrayCommonBehaviors[propertyName].call(this,instance);
+          }
+
+          if (isNaN(propertyName.toString())) {
+            return target[propertyName];
+          }
+
+          return ES6._dartArrayGet(instance,+propertyName);
+        },
+
+        'has' : function(target,prop) {
+          var len = ES6._dartArrayLength(instance);
+          var pos = +prop;
+          if (!isNaN(prop) && pos>=0 && pos < len ) {
+            return true;
+          }
+          return prop in target;
+        },
+
+        'getOwnPropertyDescriptor': function(target, prop) {
+          var len = ES6._dartArrayLength(instance);
+          var pos = +prop;
+          if (!isNaN(prop) && pos>=0 && pos < len ) {
+            return { configurable: true, enumerable: true, value: ES6._dartArrayGet(instance,pos) };
+          }
+          return Object.getOwnPropertyDescriptor(target,prop);
+        },
+
+        'set' : function(target,propertyName,value,proxy) {
+          if (propertyName.toString() == 'length') {
+            ES6._dartArraySetLength(instance,+value);
+          } else if (isNaN(propertyName.toString())) {
+            target[propertyName] = value;
+          } else {
+            ES6._dartArrayPut(instance,+propertyName,value);
+          }
+          return true;
+        },
+        'ownKeys' : function(target) {
+          var x = Object.getOwnPropertyNames(target).slice();
+          for (var i = 0 ;i<ES6._dartArrayLength(instance);i++) {
+            x.push(""+i);
+          }
+          return x;
+        }
+      });
+    };
+
+
+    /**
+     * Maps ES6JsProxy support
+     */
+
+    var mapCommonBehaviors = {
+      '__dartClass__' : function(instance) { return instance; },
+      '__isES6Proxy__' : function(instance) { return true; }
+    };
+
+    function getPropertyDescriptor(obj,prop) {
+      while(obj!=null) {
+        var pd = Object.getOwnPropertyDescriptor(obj,prop);
+        if (pd) {
+          return pd;
+        }
+        obj = obj.__proto__;
+      }
+      return undefined;
+    }
+
+    ES6.createES6JsProxyForMaps = function(instance) {
+      return new Proxy({},{
+        'has' : function(target,prop){
+          return ES6._dartKeys(instance).indexOf(prop) >= 0;
+        },
+        getOwnPropertyDescriptor: function(target, prop) {
+          if (ES6._dartKeys(instance).indexOf(prop) >= 0) {
+            return { configurable: true, enumerable: true, value: ES6._dartGet(instance,prop) };
+          }
+          return Object.getOwnPropertyDescriptor(target,prop);
+        },
+
+        'get' : function(target,propertyName,proxy) {
+
+          if (ES6._dartKeys(instance).indexOf(propertyName)>=0) {
+            return ES6._dartGet(instance,propertyName);
+          }
+
+          // target first! (because of '__proto__' , 'constructor' , and dart interop internals ... )
+          var _cache;
```

Comment:
right. I had a different logic at some point in time but now in the final version it makes no sense. just return earlier should work.